src/backend.rs

```diff
@@ -92,3 +92,50 @@ impl TryFrom<u8> for NoId {
         Err(Error::InternalError)
     }
 }
+
+/// Helper type for optional backends.
+///
+/// If the backend is `None`, [`Error::RequestNotAvailable`][] is returned.
+#[derive(Debug)]
+pub struct OptionalBackend<T>(pub Option<T>);
```

Comment:
nit: I would make the inner field private and add a `fn new` and `fn into_inner` instead.